src/libraries/System.Utf8String.Experimental/src/System/Net/Http/Utf8StringContent.cs

```diff
@@ -40,11 +42,38 @@ public Utf8StringContent(Utf8String content, string? mediaType)
         protected override Task<Stream> CreateContentReadStreamAsync() =>
             Task.FromResult<Stream>(new Utf8StringStream(_content));
 
+#if NETSTANDARD2_0
+        protected async override Task SerializeToStreamAsync(Stream stream, TransportContext? context)
+        {
+            ReadOnlyMemory<byte> buffer = _content.AsMemoryBytes();
+            if (MemoryMarshal.TryGetArray(buffer, out ArraySegment<byte> array))
+            {
+                await stream.WriteAsync(array.Array, array.Offset, array.Count).ConfigureAwait(false);
+            }
+            else
+            {
+                byte[] localBuffer = ArrayPool<byte>.Shared.Rent(buffer.Length);
+                try
+                {
+                    buffer.Span.CopyTo(localBuffer);
+                    await stream.WriteAsync(localBuffer, 0, buffer.Length).ConfigureAwait(false);
+                }
+                finally
+                {
+                    ArrayPool<byte>.Shared.Return(localBuffer);
```

Comment:
Don't put this inside a _finally_ block. Generally speaking, if an operation fails while operating on an array which has been rented from the array pool, it's safest to abandon the array and to let the GC reclaim it during the next collection. This minimizes the risk of a buggy implementation attempting to use the array (and thus corrupt global app state) after returning a faulted task.